File: backend/module-maintenance/src/main/java/univ/lille/module_maintenance/domain/model/Priority.java
package univ.lille.module_maintenance.domain.model;


public enum Priority {
    LOW,
    NORMAL,
    HIGH;

    public String getDisplayColor() {
        return switch (this) {
            case LOW -> "green";
            case NORMAL -> "blue";
            case HIGH -> "red";
        };
    }
}

File: mobile/lib/features/maintenance/data/models/priority.dart
import 'package:json_annotation/json_annotation.dart';

enum Priority {
  @JsonValue('LOW')
  low,
  @JsonValue('NORMAL')
  normal,
  @JsonValue('HIGH')
  high;

  String getDisplayColor() {
    return switch (this) {
      Priority.low => 'green',
      Priority.normal => 'blue',
      Priority.high => 'red',
    };
  }

  String getDisplayName() {
    return switch (this) {
      Priority.low => 'Basse',
      Priority.normal => 'Normale',
      Priority.high => 'Haute',
    };
  }
}